File: javascript/hello-world/hello-world.js
//
// This is a stub file for the 'Hello World' exercise. It's been provided as a
// convenience to get you started writing code faster.
// Make sure to look at hello-world.spec.js--that should give you some hints about what is
// expected here.

var HelloWorld = function() {};

HelloWorld.prototype.hello = function(input) {
var name = input;
var hello = "Hello, ";
var end = "!"
var res = "";
if (!name) {
  return "Hello, World!";
}
else {
  res = hello.concat(name);
  return res.concat(end);
}
};

module.exports = HelloWorld;

File: java/hello-world/src/main/java/HelloWorld.java
public class HelloWorld {
    public static String hello(String name) {
    	if (name == null) {
    		name = "World";
        return "Hello, " + name + "!";
    }
    else if (name == "") {
    	name = "World";
    	return "Hello, " + name + "!";
    }
    else if (name.length() == 0){
    	name = "World";
    	return "Hello, " + name + "!";
    }
    else
    {
    	return "Hello, " + name + "!";
    }
    }
}
